From 454bfa047ef1a571c104a684bb95a3c066aea66e Mon Sep 17 00:00:00 2001
From: Justin Bronder <jsbronder@gmail.com>
Date: Thu, 15 Dec 2011 14:13:44 -0500
Subject: [PATCH] rospack:  remove braindead priv dropping

Trust that the user calling this application with sudo knows what
they're doing and retain privledges.

reverts fix referenced in ros ticket #2884
---
 tools/rospack/main.cpp |   21 ---------------------
 1 files changed, 0 insertions(+), 21 deletions(-)

diff --git a/tools/rospack/main.cpp b/tools/rospack/main.cpp
index b6282e2..bd43e41 100644
--- a/tools/rospack/main.cpp
+++ b/tools/rospack/main.cpp
@@ -44,27 +44,6 @@ int main(int argc, char **argv)
     return 0;
   }
 
-#if !defined(WIN32)
-  // If it looks we're running under sudo, try to drop back to the normal
-  // user, to avoid writing the cache with inappropriate permissions,
-  // #2884.
-  // Do the group first, because we can't do it after changing the user.
-  char* sudo_gid_string = getenv("SUDO_GID");
-  if(sudo_gid_string)
-  {
-    gid_t sudo_gid = (int)strtol(sudo_gid_string, (char **)NULL, 10);
-    if(setgid(sudo_gid))
-      perror("[rospack] Failed to change GID; cache permissions may need to be adjusted manually. setgid()");
-  }
-  char* sudo_uid_string = getenv("SUDO_UID");
-  if(sudo_uid_string)
-  {
-    uid_t sudo_uid = (int)strtol(sudo_uid_string, (char **)NULL, 10);
-    if(setuid(sudo_uid))
-      perror("[rospack] Failed to change UID; cache permissions may need to be adjusted manually. setuid()");
-  }
-#endif
-
   int ret;
   bool quiet;
   try
-- 
1.7.3.4

